core/primitives/src/types.rs

```diff
@@ -119,13 +119,35 @@ pub struct FunctionCallTransaction {
     pub args: Vec<u8>,
 }
 
+#[derive(Hash, Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub struct CreateAccountTransaction {
+    pub nonce: u64,
+    pub sender: AccountId,
+    // receiver is the account id to be created
+    pub receiver: AccountId,
+    pub amount: u64,
+    pub public_key: Vec<u8>,
+}
+
+#[derive(Hash, Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub struct SwapKeyTransaction {
+    pub nonce: u64,
+    pub sender: AccountId,
+    // current key to the account.
+    // sender must sign the transaction with this key
+    pub cur_key: Vec<u8>,
```

Comment:
isn't this redundant? we will be checking that the sig matches what is on the account already